apps/arweave/src/ar_http_iface_middleware.erl

```diff
@@ -2008,21 +2008,26 @@ handle_get_chunk(OffsetBinary, Req, Encoding) ->
 								{Packing, ok};
 							{{true, _}, _StoreID} ->
 								{ok, Config} = application:get_env(arweave, config),
-								case lists:member(pack_served_chunks, Config#config.enable) of
-									false ->
-										{none, {reply, {404, #{}, <<>>, Req}}};
-									true ->
-										ok = ar_semaphore:acquire(get_and_pack_chunk,
-												infinity),
-										{RequestedPacking, ok}
+								IsPackServedChunks = lists:member(pack_served_chunks, Config#config.enable),
+								Peer = cowboy_req:peer(Req),
+								IsLocalPeerAddr = lists:foldl(fun
+									(_, true) -> true;
+									(LocalPeer, _) -> is_matching_addr(Peer, LocalPeer)
+								end, false, Config#config.local_peers),
+								case {IsPackServedChunks, IsLocalPeerAddr} of
+									{true, true} ->
+										ok = ar_semaphore:acquire(get_and_pack_chunk, infinity),
+										{RequestedPacking, ok};
+									{_, _} ->
+										{none, {reply, {403, #{}, <<>>, Req}}}
 								end
 						end,
 					case CheckRecords of
 						{reply, Reply} ->
 							Reply;
 						ok ->
 							Args = #{ packing => ReadPacking,
-									bucket_based_offset => IsBucketBasedOffset },
+									bucket_based_offset => IsBucketBasedOffset, pack => true },
```

Comment:
why are we requesting the chunk to be repacked here? This is outside the `pack_served_chunks` check, right?

Reply:
Actually no, the CheckRecords is returned from the `pack_served_chunks` clause, and repacking the chunk requres this flag.